Yosemite/YosemiteTests/Tools/POS/POSOrderServiceTests.swift

```diff
@@ -0,0 +1,105 @@
+        self.mockOrdersRemote = mockOrdersRemote
+        self.sut = POSOrderService(siteID: 123, ordersRemote: mockOrdersRemote)
+    }
+
+    @Test
+    func syncOrder_without_passing_an_order_creates_a_new_order() async throws {
+        // Given
+
+        // When
+        _ = try await sut.syncOrder(cart: [], order: nil)
+
+        // Then
+        #expect(mockOrdersRemote.createPOSOrderCalled == true)
+    }
+
+    @Test
+    func syncOrder_with_an_existing_order_updates_it() async throws {
+        // Given
+        let order = Order.fake().copy(siteID: 123, orderID: 456)
+
+        // When
+        _ = try await sut.syncOrder(cart: [], order: order)
+
+        // Then
+        #expect(mockOrdersRemote.updatePOSOrderCalled == true)
+        #expect(mockOrdersRemote.spyUpdatePOSOrder == order)
+    }
+
+    @Test func syncOrder_updates_by_deleting_existing_items_and_adding_everything_from_the_cart() async throws {
+        // Given
+        let orderItems: [OrderItem] = [
+            .fake().copy(itemID: 1, name: "Item 1", productID: 100, quantity: 1),
+            .fake().copy(itemID: 2, name: "Item 2", productID: 102, quantity: 5)]
+        let order = Order.fake().copy(siteID: 123, orderID: 456, items: orderItems)
+
+        // When
+        let cart: [POSCartItem] = [
+            makePOSCartItem(productID: 100, quantity: 2),
+            makePOSCartItem(productID: 102, quantity: 1)
+        ]
+        _ = try await sut.syncOrder(cart: cart, order: order)
+
+        // Then
+        let updatedOrderItems = try #require(mockOrdersRemote.spyUpdatePOSOrder?.items)
+        for item in updatedOrderItems {
+            #expect(item.itemID == 0 || item.quantity == 0,
+                    "Items should be deleted (quantity 0) or new (itemID 0)")
+        }
+    }
+
+    @Test func syncOrder_after_removing_item_from_cart_deletes_it_from_the_order() async throws {
+        // Given
+        let orderItems: [OrderItem] = [
+            .fake().copy(itemID: 1, name: "Item 1", productID: 100, quantity: 1),
+            .fake().copy(itemID: 2, name: "Item 2", productID: 102, quantity: 5)]
+        let order = Order.fake().copy(siteID: 123, orderID: 456, items: orderItems)
+
+        // When
+        let cart: [POSCartItem] = [
+            makePOSCartItem(productID: 102, quantity: 1)
+        ]
+        _ = try await sut.syncOrder(cart: cart, order: order)
+
+        // Then
+        let updatedOrderItems = try #require(mockOrdersRemote.spyUpdatePOSOrder?.items)
+
+        #expect(updatedOrderItems.contains(where: { item in
+            item.itemID == 1 && item.quantity == 0
+        }), "Item 1 should be deleted")
+
+        #expect(!updatedOrderItems.contains(where: { item in
+            item.productID == 100 && item.quantity > 0
+        }), "Product for item 1 should not be re-added")
+
+        #expect(updatedOrderItems.contains(where: { item in
+            item.productID == 102 && item.quantity > 0
```

Comment:
nit: maybe can ensure the item quantity is `1` as in the cart?
```suggestion
            item.productID == 102 && item.quantity == 1
```

Reply:
WDYT about #expect'ing on each item in the `updatedOrderItems` by array index? like `#expect(updatedOrderItems[0] == ...)` to ensure that every order item is as expected.

Reply:
@jaclync I did it this way because I don't want to make the tests brittle.

I don't think the sort order of the array is important, but if it is, it'll only be that deletions should come before or after additions, order within the deletions or additions could change and this test should remain valid.

The quantity could change if we alter the test data, but as long as it's being added back, that's enough for this test.